### PR TITLE
Expand settings, Implement auto-logout, auto-start features

### PR DESCRIPTION
**Pull request by sarawut11**
https://github.com/mwcproject/mwc-qt-wallet/issues/158

### CODE REVIEW

File: resource/mwcwallet_style.css [hunk 1]
```diff
@@ -659,4 +659,17 @@ QTabBar::tab:hover{
 
 QTabBar::tab:selected{
     font-weight:bold;
+}
+
+QGroupBox {
+    font: bold;
```
Comment (bayk) [inline resource/mwcwallet_style.css:665]: For light theme white color was needed. Please check if  light theme works fine
    color: white;

File: state/a_inputpassword.cpp [hunk 2]
```diff
@@ -138,8 +138,7 @@ void InputPassword::onLoginResult(bool ok) {
 
             if ( config::isOnlineWallet() && context->wallet->hasPassword() ) {
                 // Start listening, no feedback interested
-                context->wallet->listeningStart(true, false, true);
-                context->wallet->listeningStart(false, true, true);
+                context->wallet->listeningStart(context->appContext->isAutoStartMQSEnabled(), context->appContext->isAutoStartKeybaseEnabled(), true);
```
Comment (bayk) [inline state/a_inputpassword.cpp:141]: That will not work for ,true, true, case.  Sorry, interface is confusing, single call can start only one listener.
will work:
if (context->appContext->isAutoStartMQSEnabled())
     context->wallet->listeningStart(true, false, true);
if (context->appContext->isAutoStartKeybaseEnabled())
    context->wallet->listeningStart(false, true, true);

File: windows/x_walletconfig_w.cpp [hunk 3]
```diff
@@ -502,8 +554,77 @@ void WalletConfig::updateLogsStateUI(bool enabled) {
...
+        case 10 * 60:
+            ui->logout_10->setChecked(true);
+            break;
+        case 20 * 60:
+            ui->logout_20->setChecked(true);
+            break;
+        case 30 * 60:
+            ui->logout_30->setChecked(true);
+            break;
+    }
```
Comment (bayk) [inline windows/x_walletconfig_w.cpp:627]: time can be any value. Current default is 900 seconds and user always can change it in the config. We should handle that by selecting some reasonable value.  For 900 seconds you probably can select 20 minutes.

File: windows/x_walletconfig_w.cpp [hunk 4]
```diff
@@ -386,6 +400,10 @@ void WalletConfig::on_restoreDefault_clicked()
 
     updateLogsStateUI(true);
 
+    updateAutoStartStateUI(true, true);
+
+    updateAutoLogoutStateUI(3 * 60);
```
Comment (bayk) [inline windows/x_walletconfig_w.cpp:405]: Please make larger default value. Currently default for config is 15 minutes. You can make it 20.

File: windows/x_walletconfig_w.cpp [hunk 5]
```diff
@@ -437,6 +455,40 @@ bool WalletConfig::applyChanges() {
...
+                control::MessageBox::messageText(nullptr, "Internal Error",
+                                             "Unable to update wallet config file " + configFN );
+            }
+
+            bool updateOk = reader.updateConfig("logoutTimeout", QString::number(currentLogoutTimeout));
+
+            if (!updateOk) {
+                control::MessageBox::messageText(nullptr, "Error", "Wallet unable to set the selected logout time." );
+            }
+
```
Comment (bayk) [inline windows/x_walletconfig_w.cpp:484]: Also you need to change the current value for this running app. 
state::StateMachine     mwc-qt-wallet/state/statemachine.h/cpp   manage timeout,   it is using  config::getLogoutTimeMs() to read the value.
You should update   logoutTimeMs  value at mwc-qt-wallet/core/Config.cpp  to change timeout for the running app.

**Comment by jbyrer** [comment]
Hi Sarawut,

If you don't mind, please give us a more detailed description of the new functionality, and any suggestions you have regarding testing.  I want to make sure we don't miss anything.

**Pull request merged**